### ISSUE TITLE
Array notifications are applied wrong after inserting and removing a row

### ISSUE THREAD

**Issue by warpech** [Juicy/juicy-table-repeat#5]
Steps to reproduce:
1. Click on "Insert a row"
2. Click on "Update first row"
3. Click on "Remove first row"
4. Click on "Update first row"

The last 3 rows are updated with a new price. Only one row (the first row) should be updated.

See:

![screen](https://cloud.githubusercontent.com/assets/566463/17964039/802306fa-6ab8-11e6-9bfa-89ef09f5eb98.gif)

**Comment by miyconst** [comment]
@tomalec could you please take a look?

**Comment by tomalec** [comment]
Shorted test case:
1. Click on "Update first row"
2. Click on "Remove first row"
3. Click on "Update first row"

**Comment by tomalec** [comment]
It is index `.0` vs key `#0` mismatch, probably due to the fact that we are re-stamping all the rows at https://github.com/Juicy/juicy-table-repeat/blob/gh-pages/juicy-table-repeat.html#L77 in case of remove/insert of just one.

**Comment by tomalec** [comment]
It seems we are facing all the problems that are solved by Polymer's `dom-repeat` so we would probably need to move most of its logic to here.

@warpech, @miyconst WDYT about forking (and making a PR of) `dom-repeat` and implementing what was proposed at https://github.com/Polymer/polymer/issues/3448?

> [edit: I have removed my proposition as it was simply wrong]

For me injecting one more template to the dom-repeat seems easier that replicating its behavior. Also to replicate dom-repeat logic, we would need to go through and understand entire code anyway.

**Comment by tomalec** [comment]
Or maybe we should just ask people to fallback to:

```
<juicy-table>
<template is="dom-repeat">
</template>
</juicy-table>
```

na make juicy-element do nothing more than just extending `HTMLTableElement` ?

**Comment by warpech** [comment]
``` cs
<juicy-table>
<template is="dom-repeat">
</template>
</juicy-table>
```

If it works, I'd love it!

**Comment by miyconst** [comment]
@tomalec am I right that you want to stamp `<tr>`s with native Polymer's `dom-repeat` and them move them into `<table>`?

Well, if that would work, I am all for it.

P.S. Any solution which works in IE to generate table rows is good for me, even if it requires a bit more code or another `<template>`.

**Comment by warpech** [comment]
I wonder if you can get this to work:

``` html
<juicy-table class="table"><!-- "table" is a Bootstrap class -->
  <thead>
    <tr>
      <template is="dom-repeat" items="{{Headers}}"><!-- my column headers -->
        <th>{{item.Label}}</th>
      </template>
    </tr>
  </thead>
  <tbody>
    <template is="dom-repeat" items="{{People}}"><!-- my rows -->
      <tr><td>{{item.FirstName}}</td><td>{{item.LastName}}</td></tr>
    </template>
  </tbody>
</juicy-table>
```

See source of Bootstrap [`.table`](https://github.com/twbs/bootstrap/blob/v3.3.7/dist/css/bootstrap.css#L251)

**Comment by tomalec** [comment]
@warpech 

> If it works, I'd love it!

It works but you loose user agent styles

See https://jsbin.com/coxere/edit?html,output

---

@miyconst 

> am I right that you want to stamp <tr>s with native Polymer's dom-repeat and them move them into <table>?
> Not really. Anyway I need to review my proposition in Polymer's issue as it seems not to solve the problem.

**Comment by miyconst** [comment]
@warpech as far as I remember it won't work. IE removes every node from `<tbody>` and `<table>` which is not `<tr>` and puts it after the element.

So, it will be equal to:

``` html
<juicy-table class="table"><!-- "table" is a Bootstrap class -->
  <thead>
    <tr>
    </tr>
  </thead>
  <template is="dom-repeat" items="{{Headers}}"><!-- my column headers -->
    <th>{{item.Label}}</th>
  </template>
  <tbody>
  </tbody>
  <template is="dom-repeat" items="{{People}}"><!-- my rows -->
    <tr><td>{{item.FirstName}}</td><td>{{item.LastName}}</td></tr>
  </template>
</juicy-table>
```

Does not seem to work even in Chrome - [https://jsfiddle.net/qc5qv5e5/](https://jsfiddle.net/qc5qv5e5/).

**Comment by tomalec** [comment]
For that we will need `juicy-tr`, `juicy-thead`,...

**Comment by miyconst** [comment]
@tomalec your solution does not seem to work in IE:

![image](https://cloud.githubusercontent.com/assets/6435556/18056326/b9d4d9ae-6e0b-11e6-8950-4cbf7dd5dbb5.png)

**Comment by tomalec** [comment]
My bad.
You'r right, I was checking in Edge :( 

So we need to either
1. fix this bug. I'm afraid that this and bugs that are to come, will require us to do more or less this: https://github.com/Polymer/polymer/blob/master/src/lib/template/dom-repeat.html, so in long term it should be easier to..
2. Implement (few weeks) / wait for Polymer to implement (∞ ?) https://github.com/Polymer/polymer/issues/3448#issuecomment-243159847

**Comment by tomalec** [comment]
@warpech how urgently do we need a fix, and how crucial it is?

**Comment by miyconst** [comment]
@warpech, @tomalec what if we go the "smart" way and just drop the IE support?

**Comment by tomalec** [comment]
> what if we go the "smart" way and just drop the IE support?

I'd love to :sunglasses: Especially, that it seems that event Microsoft is pushing Edge forward and do not bother much about IE11 support.

The question is: what about our customers, and demos that are supposed to look clean and work cross platform?

**Comment by warpech** [comment]
No, it is not urgent. I just saw it and reported it for reference.

Let's not drop IE11support before Polymer.

**Comment by miyconst** [comment]
> Let's not drop IE11support before Polymer.

Polymer only partially supports it.

**Comment by warpech** [comment]
Let's wait until Polymer 2.0, which may fix this

**Comment by tomalec** [comment]
https://github.com/Starcounter/CompanyTrack/issues/233
